pymfe/itemset.py

```diff
@@ -0,0 +1,179 @@
+        ----------
+        C : :obj:`np.ndarray`, optional
+            Attributes from fitted data.
+
+        **kwargs
+            Additional arguments. May have previously precomputed before this
+            method from other precomputed methods, so they can help speed up
+            this precomputation.
+
+        Returns
+        -------
+        :obj:`dict`
+            With following precomputed items:
+                - ``itemset_binary_matrix`` (:obj:`list`): Binary
+                  representation of the attributes.
+        """
+        prepcomp_vals = {}
+        if C is not None and 'itemset_binary_matrix' not in kwargs:
+            itemset_binary_matrix = MFEItemset._matrix_to_binary(C)
+            prepcomp_vals["itemset_binary_matrix"] = itemset_binary_matrix
+        return prepcomp_vals
+
+    @staticmethod
+    def _array_to_binary(array: np.ndarray):
+        """Convert an array to its binary representation."""
+        values = np.unique(array)
+        res = np.zeros((array.shape[0], values.shape[0])).astype(bool)
+        for i, val in enumerate(values):
+            res[:, i] = array == val
+        return res
+
+    @staticmethod
+    def _matrix_to_binary(C: np.array):
+        """Convert an matrix to its binary representation."""
+        return [MFEItemset._array_to_binary(col) for col in C.T]
+
+    @classmethod
+    def ft_two_itemset(cls,
+                       C: np.ndarray,
+                       itemset_binary_matrix: t.List[np.ndarray] = None,
+                       ) -> np.ndarray:
+        """Computes the one itemset meta-feature, which is the individual
+        frequency of each attribute in binary format.
+
+        Parameters
+        ----------
+        N : :obj:`np.ndarray`
+            Attributes from fitted data.
+
+        itemset_binary_matrix : list
+            Binary representation of the attributes. Each list value has a
+            binary representation of each attributes in the dataset.
+
+        Returns
+        -------
+        :obj:`np.ndarray`
+            An array with the twoitem for each attribute.
+
+        References
+        ----------
+        .. [1] Song, Q., Wang, G., & Wang, C. (2012). Automatic recommendation
+           of classification algorithms based on data set characteristics.
+           Pattern recognition, 45(7), 2672-2689.
+        """
+        if itemset_binary_matrix is None:
+            sub_dic = MFEItemset.precompute_binary_matrix(C)
+            itemset_binary_matrix = sub_dic["itemset_binary_matrix"]
+
+        B = itemset_binary_matrix
+
+        result = []  # type: t.List[float]
+        while B:
+            Bi = B[0]
+            del B[0]
+            for Bj in B:
+                aux = [np.sum(np.logical_xor(i, j))
+                       for i in Bi.T for j in Bj.T]
+                result += aux
+
+        twoitem_by_attr = np.array(result) / C.shape[0]
+
+        return twoitem_by_attr
+
+    @classmethod
+    def ft_one_itemset(cls,
```

Comment:
The metafeature name says "one_itemset" metafeature but the pydoc says "two itemset metafeature".

Reply:
Yes, it is wrong. I will solve it. Thank You.